shared/github/__init__.py

```diff
@@ -48,6 +54,32 @@ def __init__(self, error_cause: InstallationErrorCause, *args: object) -> None:
         self.error_cause = error_cause
 
 
+def decide_installation_error_cause(
+    response: requests.Response,
+) -> InstallationErrorCause:
+    # int(res.headers.get("X-RateLimit-Remaining", -1)) == 0
+    is_primary_rate_limit = int(response.headers.get("X-RateLimit-Remaining", -1)) == 0
+    is_secondary_rate_limit = response.headers.get("Retry-After") is not None
+    match response.status_code:
+        case 404:
+            return "installation_not_found"
+        case 429:
```

Comment:
this is unreachable, as you call the function within a `status_code in (404, 403)` block

Reply:
Thanks for pointing it out